Gemfile

```diff
@@ -37,11 +33,18 @@ end
 
 group :test do
   gem "capybara", ">= 2.15"
+  gem "database_cleaner", "~> 1.5"
   gem "selenium-webdriver"
+  gem "shoulda-matchers", "~> 3.0", require: false
+  gem "simplecov"
+  gem "simplecov-rcov"
```

Comment:
Các gem này có tác dụng gì e

Reply:
1.   gem "database_cleaner":  sau khi chạy xong một test nó sẽ rollback database lại trạng thái trước khi chạy test, đảm bảo tinsh nhất quán trong quá trình thực hiện tất cả các test
2.  gem "shoulda-matchers": gem này giúp  viết test dễ dàng hơn, tiết kiệm được thời gian khi viết các test dài và phức tạp, code ngắn gọn, dễ đọc.
3.  gem "simplecov": gem này giúp hỗ trợ tinsh toán những dòng nào được cover test, dòng nào chưa
4. gem "simplecov-rcov": style formatter  cho simplecov